### PR TITLE
Add standard zero centred normalization for SGDRegressor

### PR DESCRIPTION
**Pull request by bestetc**
Fixed some rare cases where `calc_init_by_layers` would return an incorrect evaluation for `init`. @nikita-klsh suggested using the zero-centered normalization after splitting data into layers instead of using normalization to [0, 1] interval before splitting the data.

We did a the small reasearch how different types of normalization and hyperparameters for SGDRegressor affect the calculation of "initialization", and chose the best of the stable options.

This pull reqest
* fix the bad cases
bad_fit

* good cases still good cases
same_fit

* makes RefractorVelocity fit a little bit faster
timing

### CODE REVIEW

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 1]
```diff
@@ -401,34 +407,34 @@ def _calc_init_by_layers(self, n_refractors):
...
+            mask = (self.offsets > cross_offsets[i]) & (self.offsets <= cross_offsets[i + 1])
             if mask.sum() > 1:  # at least two point to fit
-                current_slope[i], current_time[i] = \
-                    self._fit_regressor(normalized_offsets[mask].reshape(-1, 1), normalized_times[mask],
-                                        initial_slope, initial_time)
+                # data normalization occurs independently for each layer
+                scaled_offsets, mean_offset, std_offset = self._standart_scaler(self.offsets[mask])
+                scaled_times, mean_time, std_time = self._standart_scaler(self.fb_times[mask])
+                fitted_slope, fitted_time = self._fit_regressor(scaled_offsets.reshape(-1, 1), scaled_times,
+                                                                initial_slope * std_offset / std_time, 0)
```
Comment (nikita-klsh) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:424]: Is not it?

```suggestion
                fitted_slope, fitted_time = self._fit_regressor(scaled_offsets.reshape(-1, 1), scaled_times, 1, 0)
```

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 2]
```diff
@@ -492,6 +511,67 @@ def _ms_to_kms(self, params, as_array=True):
...
+            raise ValueError("`coords` attribute should be defined.")
+        col_size = max(col_size, max(len(name) for name in self.coords.names) + 1)
+        values = list(self.coords.coords) + list(self.params.values())
+        values_format = '\n' + '{:>{col_size}}' * len(self.coords) + '{:>{col_size}.2f}' * len(self.params)
+
+        columns = list(self.coords.names) + list(self.params.keys())
+        cols_format = '{:>{col_size}}' * (len(columns))
+        with open(path, 'w', encoding=encoding) as f:
+            f.write(cols_format.format(*columns, col_size=col_size))
+            f.write(values_format.format(*values, col_size=col_size))
```
Comment (anton-br) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:550]:
```suggestion
            values = list(self.coords.coords) + list(self.params.values())
            values_format = '\n' + '{:>{col_size}}' * len(self.coords) + '{:>{col_size}.2f}' * len(self.params)
            values_format = values_format.format(*values, col_size=col_size)
            
            columns = list(self.coords.names) + list(self.params.keys())
            cols_format = ()'{:>{col_size}}' * (len(columns)).format(*columns, col_size=col_size)
            with open(path, 'w', encoding=encoding) as f:
                f.write(cols_format + values_format)
```

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 3]
```diff
@@ -492,6 +511,67 @@ def _ms_to_kms(self, params, as_array=True):
             return values
         return dict(zip(self._valid_keys, values))
 
+    def dump(self, path, col_size=10, encoding="UTF-8"):
+        """Dump RefractorVelocity instance to disk.
```
Comment (anton-br) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:515]:
```suggestion
        """Save the RefractorVelocity instance to a file.
```

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 4]
```diff
@@ -492,6 +511,67 @@ def _ms_to_kms(self, params, as_array=True):
...
+
+        File contains coords and params attributes with follow structure:
+        [coords.names] [params.keys]
+        [coords.coords] [params.values]
+
+        Parameters
+        ----------
+        path : str,
+            path to the file with parameters.
```
Comment (anton-br) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:524]: I've stolen this from gather.dump, seems better for me.
Also, make sure that the sentences start with capital letters
```suggestion
            The directory to dump the RefractorVelocity instance in.
```

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 5]
```diff
@@ -492,6 +511,67 @@ def _ms_to_kms(self, params, as_array=True):
...
+            RefractorVelocity without changes.
+
+        Raises
+        ------
+        ValueError
+            If coords attributes is None.
+        """
+        if self.coords is None:
+            raise ValueError("`coords` attribute should be defined.")
+        col_size = max(col_size, max(len(name) for name in self.coords.names) + 1)
```
Comment (anton-br) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:542]: Do you really need to change the size of all columns if the coords.name is longer than col_size? May be change only the coords-related columns, leaving other columns' size to col_size?

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 6]
```diff
@@ -492,6 +511,67 @@ def _ms_to_kms(self, params, as_array=True):
...
+
+        columns = list(self.coords.names) + list(self.params.keys())
+        cols_format = '{:>{col_size}}' * (len(columns))
+        with open(path, 'w', encoding=encoding) as f:
+            f.write(cols_format.format(*columns, col_size=col_size))
+            f.write(values_format.format(*values, col_size=col_size))
+        return self
+
+    def load(self, path, encoding="UTF-8"):
+        """Load parameters from a file and create a RefractorVelocity instance from the loaded parameters.
```
Comment (anton-br) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:554]: simplify the sentence, you have two `parameters` here.
Also, it is not clear what parameters do you load

**Review by anton-br — COMMENTED** [review]
Overall, do you really need to be able to dump single RV?? Probably you want to dump only RV field? The dump and load you wrote here are easelly expanding for many RV.

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 7]
```diff
@@ -338,7 +357,7 @@ def loss_piecewise_linear(self, args, loss='L1', huber_coef=20):
     def _get_valid_keys(self, n_refractors=None):
         """Returns a list with the valid keys based on `n_refractors` or `self.n_refractors`."""
         n_refractors = self.n_refractors if n_refractors is None else n_refractors
-        return ['t0'] + [f'x{i + 1}' for i in range(n_refractors - 1)] + [f'v{i + 1}' for i in range(n_refractors)]
+        return ['t0'] + [f'x{i}' for i in range(1, n_refractors)] + [f'v{i + 1}' for i in range(n_refractors)]
```
[the remark this answers is not in the document]
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py:360]: a little bit less symbols.
second will be `[f'v{i}' for i in range(1, n_refractors + 1)]`. It's a little more symbols.

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 4 above)
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py]: It's full path to a file

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 8]
```diff
@@ -401,34 +409,28 @@ def _calc_init_by_layers(self, n_refractors):
...
+            scaled_times, mean_time, std_time = self._standart_scaler(self.fb_times[mask])
+            if std_offset not in [0, np.nan] and std_time not in [0, np.nan]:
+                fitted_slope, fitted_time = self._fit_regressor(scaled_offsets.reshape(-1, 1), scaled_times, 1, 0)
+                current_slope[i] = fitted_slope * std_time / std_offset
+                current_time[i] = mean_time + fitted_time * std_time - current_slope[i] * mean_offset
             else:
-                current_slope[i] = initial_slope
-                current_time[i] = initial_time
+                # raise base velocity for the next layer (v = 1 / slope)
+                current_slope[i] = current_slope[i] * (n_refractors / (n_refractors + 1)) if i else 4 / 5
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:427]: But it's initialized empty in the rhs

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 9]
```diff
@@ -401,34 +409,28 @@ def _calc_init_by_layers(self, n_refractors):
...
-            current_slope[i] = max(.167 * self.max_offset / max_fb_times, current_slope[i])
+            current_slope[i] = max(.167, current_slope[i])
             current_time[i] =  max(0, current_time[i])
-            # raise base velocity for the next layer (v = 1 / slope)
-            initial_slope = current_slope[i] * (n_refractors / (n_refractors + 1))
-            initial_time = current_time[i] + (current_slope[i] - initial_slope) * cross_offsets[i + 1]
-        velocities = 1 / (current_slope * max_fb_times / self.max_offset)
-        init = [current_time[0] * max_fb_times, *cross_offsets[1:-1] * self.max_offset, *(velocities * 1000)]
+        velocities = 1 / current_slope
+        init = [current_time[0], *cross_offsets[1:-1], *(velocities * 1000)]
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:433]: Velocities can be non-increasing here

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 10]
```diff
@@ -178,7 +178,7 @@ def from_first_breaks(cls, offsets, fb_times, init=None, bounds=None, n_refracto
 
         # fitting piecewise linear regression
         partial_loss_func = partial(self.loss_piecewise_linear, loss=kwargs.pop('loss', 'L1'),
-                                    huber_coef=kwargs.pop('huber_coef', .1))
+                                    huber_coef=kwargs.pop('huber_coef', 20))
         minimizer_kwargs = {'method': 'SLSQP', 'constraints': constraints_list, **kwargs}
         self._model_params = optimize.minimize(partial_loss_func, x0=self._ms_to_kms(self.init),
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:183]: `piecewise_times` and `piecewise_offsets` are updated on each optimization step inside `partial_loss_func`, while `_model_params` contains parameters of the best iteration, which can be not the last one. In this case `self.params` and `piecewise_*` become out of sync

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 11]
```diff
@@ -357,29 +364,14 @@ def _get_constraints(self):
...
-                               max_iter=1000, learning_rate='optimal')
-        lin_reg.fit(x, y, coef_init=start_slope, intercept_init=start_time)
-        return lin_reg.coef_[0], lin_reg.intercept_
+    @staticmethod
+    def _scale_standart(data):
+        """Scale data to zero mean and unit variance."""
+        if len(data) == 0:
+            return data, np.nan, np.nan
+        mean, std = np.mean(data), np.std(data)
+        data_scaled = (data - mean) / (std + 1e-10)
```
Comment (dpodvyaznikov) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:373]: I'd suggest using `normalization.scale_standard`:

```suggestion
        data_scaled = normalization.scale_standard(data, mean, std, 1e-10)
```

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 12]
```diff
@@ -357,29 +364,14 @@ def _get_constraints(self):
...
-        -------
-        params : tuple
-            Linear regression `coef` and `intercept`.
-        """
-        lin_reg = SGDRegressor(loss='huber', penalty=None, shuffle=True, epsilon=0.05, eta0=0.1, alpha=0.01, tol=1e-6,
-                               max_iter=1000, learning_rate='optimal')
-        lin_reg.fit(x, y, coef_init=start_slope, intercept_init=start_time)
-        return lin_reg.coef_[0], lin_reg.intercept_
+    @staticmethod
+    def _scale_standart(data):
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:368]:
```suggestion
    def _scale_standard(data):
```

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]
1. Remove `_update_piecewise_coords` and `_create_piecewise_coords` completely
2. Make a new **static** method `calc_knots_by_params` that takes **only params** and returns offsets and times of knots of the piecewise-linear velocity function
3. Call it in `loss_piecewise_linear` and in `from_first_breaks` after the fit

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 11 above)
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py]: Also move normalization from `gather.utils` to `utils`

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 13]
```diff
@@ -262,15 +257,11 @@ def has_coords(self):
         """bool: Whether refractor velocity coordinates are not-None."""
         return self.coords is not None
 
-    def _create_piecewise_coords(self, n_refractors, max_offset=np.nan):
-        """Create two array corresponding to the piecewise linear function coords."""
+    def calc_knots_by_params(self, params, n_refractors, max_offset):
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:260]: I asked you to make it static. You don't need `n_refractors` here, they can be calculated from params

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 14]
```diff
@@ -262,15 +257,11 @@ def has_coords(self):
...
+        """Update the given `offsets` and `times` arrays based on the `params` and `n_refractors`."""
         piecewise_offsets = np.zeros(n_refractors + 1)
         piecewise_times = np.zeros(n_refractors + 1)
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:263]: Why not `empty`?

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 15]
```diff
@@ -317,18 +308,18 @@ def loss_piecewise_linear(self, args, loss='L1', huber_coef=20):
             If given `loss` does not exist.
         """
         self.piecewise_offsets, self.piecewise_times = \
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:310]: Don't update `self` here

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 16]
```diff
@@ -348,38 +339,16 @@ def _get_constraints(self):
...
-                               max_iter=1000, learning_rate='optimal')
-        lin_reg.fit(x, y, coef_init=start_slope, intercept_init=start_time)
-        return lin_reg.coef_[0], lin_reg.intercept_
+    @staticmethod
+    def _scale_standard(data):
+        """Scale data to zero mean and unit variance."""
+        if len(data) == 0:
+            return data, np.nan, np.nan
+        mean, std = np.mean(data), np.std(data)
+        data_scaled = normalization.scale_standard(data, mean, std, 1e-10)
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:350]: I'd stick to the previous version, `normalization` module was designed to perform inplace normalization whenever possible. `scale_standard` is the only exception now, but may change in the future.

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 17]
```diff
@@ -401,34 +370,29 @@ def _calc_init_by_layers(self, n_refractors):
...
+                lin_reg.fit(scaled_offsets.reshape(-1, 1), scaled_times, coef_init=1, intercept_init=0)
+                current_slope[i] = lin_reg.coef_[0] * std_time / std_offset
+                current_time[i] = mean_time + lin_reg.intercept_ * std_time - current_slope[i] * mean_offset
             else:
-                current_slope[i] = initial_slope
-                current_time[i] = initial_time
-            # move maximal velocity to 6 km/s
-            current_slope[i] = max(.167 * self.max_offset / max_fb_times, current_slope[i])
+                # raise base velocity for the next layer (v = 1 / slope) and use default times (defaults to 0)
+                current_slope[i] = current_slope[i - 1] * (n_refractors / (n_refractors + 1)) if i else 4 / 5
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:390]: Don't get why `current_time` is not updated here. Maybe simply use slope and time from the previous layer?

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 18]
```diff
@@ -401,34 +370,29 @@ def _calc_init_by_layers(self, n_refractors):
...
         current_slope = np.empty(n_refractors)
-        current_time = np.empty(n_refractors)
-
-        for i in range(n_refractors):  # inside the loop work with normalized data only
-            mask = (normalized_offsets > cross_offsets[i]) & (normalized_offsets <= cross_offsets[i + 1])
-            if mask.sum() > 1:  # at least two point to fit
-                current_slope[i], current_time[i] = \
-                    self._fit_regressor(normalized_offsets[mask].reshape(-1, 1), normalized_times[mask],
-                                        initial_slope, initial_time)
+        current_time = np.zeros(n_refractors)
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:375]: Why do you need an array of current times if you use only the first one?

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]
`_empty_layers` is unused now. `_ms_to_kms` uses default value of `to_array` all the time.

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 17 above)
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py:390]: defaults params for t0 is 0, if no data found (or this data could not be normalize) current_time still set to init values. 
init values for time is 0.

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 18 above)
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py:375]: change `np.zeros` to `0`.

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 14 above)
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py:264]: changed to 0

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 19]
```diff
@@ -401,34 +370,30 @@ def _calc_init_by_layers(self, n_refractors):
...
-            # raise base velocity for the next layer (v = 1 / slope)
-            initial_slope = current_slope[i] * (n_refractors / (n_refractors + 1))
-            initial_time = current_time[i] + (current_slope[i] - initial_slope) * cross_offsets[i + 1]
-        velocities = 1 / (current_slope * max_fb_times / self.max_offset)
-        init = [current_time[0] * max_fb_times, *cross_offsets[1:-1] * self.max_offset, *(velocities * 1000)]
+                # raise base velocity for the next layer (v = 1 / slope) and leave `current_time` unchanged
+                current_slope[i] = current_slope[i - 1] * (n_refractors / (n_refractors + 1)) if i else 4 / 5
+            current_slope[i] = max(.167, current_slope[i])  # move maximal velocity to 6 km/s
+
+        current_time =  max(0, current_time)
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:394]: Move it in the loop

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 20]
```diff
@@ -401,34 +370,30 @@ def _calc_init_by_layers(self, n_refractors):
...
         current_slope = np.empty(n_refractors)
-        current_time = np.empty(n_refractors)
-
-        for i in range(n_refractors):  # inside the loop work with normalized data only
-            mask = (normalized_offsets > cross_offsets[i]) & (normalized_offsets <= cross_offsets[i + 1])
-            if mask.sum() > 1:  # at least two point to fit
-                current_slope[i], current_time[i] = \
-                    self._fit_regressor(normalized_offsets[mask].reshape(-1, 1), normalized_times[mask],
-                                        initial_slope, initial_time)
+        current_time = 0
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:375]: Why `current_`? Its `t0`

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 21]
```diff
@@ -401,34 +370,30 @@ def _calc_init_by_layers(self, n_refractors):
...
-            # move maximal velocity to 6 km/s
-            current_slope[i] = max(.167 * self.max_offset / max_fb_times, current_slope[i])
-            current_time[i] =  max(0, current_time[i])
-            # raise base velocity for the next layer (v = 1 / slope)
-            initial_slope = current_slope[i] * (n_refractors / (n_refractors + 1))
-            initial_time = current_time[i] + (current_slope[i] - initial_slope) * cross_offsets[i + 1]
-        velocities = 1 / (current_slope * max_fb_times / self.max_offset)
-        init = [current_time[0] * max_fb_times, *cross_offsets[1:-1] * self.max_offset, *(velocities * 1000)]
+                # raise base velocity for the next layer (v = 1 / slope) and leave `current_time` unchanged
+                current_slope[i] = current_slope[i - 1] * (n_refractors / (n_refractors + 1)) if i else 4 / 5
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:391]: Make this a one-liner before the loop instead of instantiation using `empty`

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 20 above)
Comment (bestetc) [reply seismicpro/src/refractor_velocity/refractor_velocity.py:375]: `init_t0` is OK?

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 22]
```diff
@@ -262,15 +256,13 @@ def has_coords(self):
         """bool: Whether refractor velocity coordinates are not-None."""
         return self.coords is not None
 
-    def _create_piecewise_coords(self, n_refractors, max_offset=np.nan):
-        """Create two array corresponding to the piecewise linear function coords."""
+    @staticmethod
+    def calc_knots_by_params(params, max_offset):
+        """Update the given `offsets` and `times` arrays based on the `params` and `n_refractors`."""
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:261]: Update docs

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 23]
```diff
@@ -281,7 +273,7 @@ def _update_piecewise_coords(self, piecewise_offsets, piecewise_times, params, n
     def loss_piecewise_linear(self, args, loss='L1', huber_coef=20):
         """Update the piecewise linear attributes and returns the loss function result.
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:274]: It only calculates the loss

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 24]
```diff
@@ -348,38 +339,16 @@ def _get_constraints(self):
         constraint_velocity = {  # velocities ascend.
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:339]: Use `append` kwarg in the above line, can't comment on it

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 25]
```diff
@@ -401,34 +370,30 @@ def _calc_init_by_layers(self, n_refractors):
...
-                    self._fit_regressor(normalized_offsets[mask].reshape(-1, 1), normalized_times[mask],
-                                        initial_slope, initial_time)
+        init_t0 = 0
+
+        for i in range(n_refractors):
+            mask = (self.offsets > cross_offsets[i]) & (self.offsets <= cross_offsets[i + 1])
+            # data normalization occurs independently for each layer
+            scaled_offsets, mean_offset, std_offset = self._scale_standard(self.offsets[mask])
+            scaled_times, mean_time, std_time = self._scale_standard(self.fb_times[mask])
+            if std_offset not in [0, np.nan] and std_time not in [0, np.nan]:
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:382]:
```suggestion
            if std_offset and std_time:
```
And return 0 std in `_scale_standard` for empty data

File: seismicpro/src/refractor_velocity/refractor_velocity.py (hunk 21 above)
Comment (alexanderkuvaev) [reply seismicpro/src/refractor_velocity/refractor_velocity.py:392]: Still unresolved

File: seismicpro/src/refractor_velocity/refractor_velocity.py [hunk 26]
```diff
@@ -401,34 +370,30 @@ def _calc_init_by_layers(self, n_refractors):
...
+                lin_reg = SGDRegressor(loss='huber', penalty=None, shuffle=True, epsilon=.1, eta0=0.1, alpha=0.01,
+                                       tol=1e-6, max_iter=1000, learning_rate='optimal')
+                lin_reg.fit(scaled_offsets.reshape(-1, 1), scaled_times, coef_init=1, intercept_init=0)
+                current_slope[i] = lin_reg.coef_[0] * std_time / std_offset
```
Comment (alexanderkuvaev) [inline seismicpro/src/refractor_velocity/refractor_velocity.py:386]: Call it `slopes`

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]

**Review by alexanderkuvaev — APPROVED** [review]

**Pull request merged**